pkg/filter/network/streamproxy/streamproxy.go

```diff
@@ -139,31 +145,46 @@ func (p *proxy) initializeUpstreamConnection() api.FilterStatus {
 		ctx:     p.ctx,
 		cluster: clusterInfo,
 	}
-	connectionData := p.getUpstreamConnection(ctx, clusterSnapshot)
-	if connectionData.Connection == nil {
-		p.requestInfo.SetResponseFlag(api.NoHealthyUpstream)
-		p.onInitFailure(NoHealthyUpstream)
 
-		return api.Stop
+	retryTime := clusterSnapshot.HostSet().Size()
+	if retryTime > 3 {
```

Comment:
better use a const variable instead of hardcode number 3? make the number readable.